fendermint/testing/contract-test/tests/upgrades.rs

```diff
@@ -0,0 +1,300 @@
+            .with_block_hash(block_hash);
+
+        self.put_exec_state(state).await;
+
+        let res = self
+            .modify_exec_state(|s| self.interpreter.begin(s))
+            .await
+            .unwrap();
+        println!("Ret begin apply_ret : {:?}", res.apply_ret);
+
+        Ok(())
+    }
+
+    async fn end_block(&self, _block_height: ChainEpoch) -> Result<()> {
+        let _ret = self
+            .modify_exec_state(|s| self.interpreter.end(s))
+            .await
+            .context("end failed")?;
+
+        Ok(())
+    }
+
+    async fn commit(&mut self) -> Result<()> {
+        let exec_state = self.take_exec_state().await;
+
+        let (
+            state_root,
+            FvmUpdatableParams {
+                power_scale,
+                circ_supply,
+            },
+            _,
+        ) = exec_state.commit().context("failed to commit FVM")?;
+
+        self.state_params.state_root = state_root;
+        self.state_params.power_scale = power_scale;
+        self.state_params.circ_supply = circ_supply;
+
+        Ok(())
+    }
+}
+
+#[test]
+fn testest() {
+    let rt: tokio::runtime::Runtime =
+        tokio::runtime::Runtime::new().expect("create tokio runtime for init");
+
+    let mut us = UpgradeScheduler::default();
+
+    let chain_id = 1942764459484029;
+
+    us.add(Upgrade {
+        chain_id,
+        block_height: 1,
+        migration: |state| {
+            println!(
+                "Migration at height {} just prints out chainmetadata actor",
+                state.block_height()
+            );
+
+            let actor = state
+                .state_tree_mut()
+                .get_actor(system::SYSTEM_ACTOR_ID)
+                .unwrap();
+            print!("chainmetadata actor: {:?}", actor);
+            Ok(())
+        },
+    })
+    .unwrap();
+
+    us.add(Upgrade {
+        chain_id,
+        block_height: 2,
+        migration: |state| {
+            println!(
+                "Migration at height {} deletes chainmetadata actor",
+                state.block_height()
+            );
+
+            state.state_tree_mut().delete_actor(system::SYSTEM_ACTOR_ID);
+
+            Ok(())
+        },
+    })
+    .unwrap();
+
+    us.add(Upgrade {
+        chain_id,
+        block_height: 3,
+        migration: |state| {
+            println!(
+                "Migration at height {} confirms the chainmetadata actor is deleted",
+                state.block_height()
+            );
+
+            let actor = state
+                .state_tree_mut()
+                .get_actor(system::SYSTEM_ACTOR_ID)
+                .unwrap();
+            print!("chainmetadata actor after delete: {:?}", actor);
+            Ok(())
+        },
+    })
+    .unwrap();
+
+    let interpreter: FvmMessageInterpreter<MemoryBlockstore, HttpClient> =
+        FvmMessageInterpreter::new(
+            tendermint_rpc::HttpClient::new("http://127.0.0.1:26657").unwrap(),
```

Comment:
Can you use the `MockClient` like I do in other tests? It would be more obvious that we don't expect CometBFT to run. Which we don't, right?

Reply:
Correct, but the issue is that the MockClient is not cloneable which is AFAICT required by the FvmExecState<MemoryBlockstore> contraint being clonable. Probably there is some way around that, need to check better

Reply:
But it's not part of the execution state, it's only a dependency of the interpreter, which stays the same throughout the execution.